### PR TITLE
Added ability to update a submodule

### PR DESCRIPTION
**Pull request by stephaneyfx**
This wraps `git_submodule_update`.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Looks great, thanks!

**Comment by stephaneyfx** [comment]
That was fast, thank you!

Do you think implementing `Repository::clone_recurse` to clone and update all submodules would make sense (basically `git clone --recursive`)?

**Comment by alexcrichton** [comment]
Yeah seems reasonable to me!

**Comment by stephaneyfx** [comment]
Thank you. I will look into it.

**Comment by stephaneyfx** [comment]
After looking into it more, I think adding an equivalent of `git clone --recursive` requires more thought and design, especially to do it in parallel like the `--jobs` option. This would need new callback types that are `Send`. Even without making it parallel, some questions don't have an obvious answer. For example, `CheckoutBuilder` has the ability to specify the directory to check out to and if `clone_recurse` takes a single instance of it, it is not clear to me how this instance can be used to check out multiple submodules.

In summary, this may need more thought to come up with a versatile and elegant design and I have other priorities at the moment. Hopefully, having `git_submodule_update` already helps, combined with the existing submodule functionality.

**Comment by alexcrichton** [comment]
Ok no worries!
